### DISCUSSION TITLE
Merging files with different normalizations and resolutions into a single .hic/.mcool file

### DISCUSSION THREAD

**Discussion by Nuturetree** [paulsengroup/hictk#306] [Q&A] [answered]
Hi 
Author
   How merged the .hic/.cool files with different resolutions and different normalization
Wariting your reply
Nututures

**Comment by robomics** [comment, marked as answer]
Hi!

We do not provide such functionality because it would not be trivial to prevent users from creating broken files by providing incompatible inputs to hictk.

Can you elaborate a bit on what your use-case for such functionality would be?
Is it something that cannot be achieved with a combination of `hictk load/merge zoomify and balance`?

**Comment by Nuturetree** [comment]
Thank you for your quick response. Let me explain the actual problem I need to solve in detail. Currently, 

I have generated 5kb (A_5kb.hic), 10kb (A_10kb.hic), and 20kb (A_20kb.hic) hic files using hictk. Additionally, I have normalized these hic files, resulting in files such as A_5kb_ice.hic, A_10kb_ice.hic, A_20kb_ice.hic, A_5kb_scale.hic, A_10kb_scale.hic, and A_20kb_scale.hic.

May I ask, how can I use hictk's merge functionality to combine these 9 files into a single hic file?

**Comment by robomics** [comment]
Thanks for providing more details.

I think what you are trying to do can be achieved as follows:
- Generate the base resolution with `hictk load`
- Create additional resolutions by iterative coarsening with `hictk zoomify`
- Balance the multi-resolution .hic file generated by the previous step with `hictk balance`
- Profit!

In practice:

```bash
hictk load interactions.pairs.gz -f 4dn A_5kb.hic --bin-size 5000  #  This you already have
hictk zoomify A_5kb.hic A_multires.hic --resolutions 5000 10000 20000  # You can also not specify resolutions and let hictk generate some default ones for you (see docs for `--nice-steps` and` --pow2-steps`)
hictk balance ice A_multires.hic --force  # --force is needed because you have already balanced the base resolution, and so you need to tell hictk to overwrite existing weights
```

`hictk merge` is only meant to be used to merge the raw counts from .cool and .hic files that are using the same reference genome.

**Comment by Nuturetree** [comment]
Can hictk generate a .hic file that contains ICE, scale, and raw statues？

**Comment by robomics** [comment]
Yes it can!

You can balance the same file (regardless of the format) with different algorithms (and even different variants of the same algorithm).

In your specific case, simply run `hictk balance scale A_multires.hic` after balancing with ICE as shown in my previous message.
There are various way to speed up loading/balance/zoomify using e.g. multiple CPU cores, more ram and/or fater IO for tempoary files.
Make sure to check out the docs for more details (e.g. the docs for `hictk balance` are available [here](https://hictk.readthedocs.io/en/stable/balancing_matrices.html)).
